functions/src/api/gemini.api.ts

```diff
@@ -127,19 +122,34 @@ export async function callGemini(
   const result = await model.generateContent(prompt);
   const response = await result.response;
 
-  if (!response || !response.candidates) {
-    console.error('Error: No response');
-    return {text: ''};
+  if (response.promptFeedback) {
+    return {
+      status: ModelResponseStatus.REFUSAL_ERROR,
+      errorMessage:
+        response.promptFeedback.blockReasonMessage ??
+        JSON.stringify(response.promptFeedback),
+    };
+  }
+
+  if (!response.candidates) {
+    return {
+      status: ModelResponseStatus.UNKNOWN_ERROR,
+      errorMessage: `Response unexpectedly had no candidates: ${response}`,
```

Comment:
Nit: Make this error more clear (experimenters might not know what candidates are)

Reply:
How about this? Experimenters probably won't be able to deal with this error anyway since it means the provider isn't returning the expected format, but hopefully this will make it clearer that this is probably a provider/platform problem.